src/rgw/rgw-gap-list

```diff
@@ -400,9 +400,12 @@ BEGIN {
   f1_count++
   if(f2_eof==0) {
     if(test_lines()==2) {
-      while ($1>b[1]) {
+      while ($1>b[1] && !f2_eof) {
```

Comment:
...and, indeed, advance_f2() can produce a side-effect on f2_eof. It might be nice to rework the logic, but unclear exactly how (perhaps advance_f2() can return when f2_eof would be set?), or whether that would really be enough improvement on its own. Do other bugs lurk within..?

Reply:
Appreciate the detailed feedback, @chardan . There's some history here. When I wrote rgw-orphan-list we needed to diff two files that were lexically sorted. Regular diff was taking a long time as the files were *very* large. So I wrote `src/tools/ceph-diff-sorted.cc` to do an optimized diff on sorted data. @linuxkidd wrote rgw-gap-list and wanted to keep the script self-contained, so did the algorithm in `awk` embedded in the shell script.

Reply:
Much appreciated, thank you! LGTM.